src/main_benchmark.cpp

```diff
@@ -0,0 +1,411 @@
+    }
+    auto workloadEnd = std::chrono::high_resolution_clock::now();
+    wholeThreadTimeUs = std::chrono::duration_cast<std::chrono::microseconds>(workloadEnd - workloadStart).count();
+    auto result = std::accumulate(latenciesWhole.begin(), latenciesWhole.end(), 0);
+    averageWholeLatency = std::accumulate(latenciesWhole.begin(), latenciesWhole.end(), 0) / (double(niterPerThread) * 1'000);
+    averagePureLatency = std::accumulate(latenciesPure.begin(), latenciesPure.end(), 0) / (double(niterPerThread) * 1'000);
+}
+}  // namespace
+// TODO support more than 1 input
+// TODO support 3 modes:
+// * fair play -> don't recreate request, but set buffer inputs
+// * full request prepare -> recreate request
+// * inference only -> dont reset buffers
+// TODO change plugin_config
+// TODO change nireq in model
+
+int main(int argc, char** argv) {
+    installSignalHandlers();
+    BenchmarkCLIParser cliparser;
+    cliparser.parse(argc, argv);
+
+    OVMS_ServerGeneralOptions* go = 0;
+    OVMS_ServerMultiModelOptions* mmo = 0;
+    OVMS_Server* srv;
+
+    OVMS_ServerGeneralOptionsNew(&go);
+    OVMS_ServerMultiModelOptionsNew(&mmo);
+    OVMS_ServerNew(&srv);
+
+    uint32_t grpcPort = cliparser.result->operator[]("port").as<uint32_t>();
+    uint32_t restPort = cliparser.result->operator[]("rest_port").as<uint32_t>();
+    OVMS_ServerGeneralOptionsSetGrpcPort(go, grpcPort);
+    OVMS_ServerGeneralOptionsSetRestPort(go, restPort);
+
+    std::string cliLogLevel(cliparser.result->operator[]("log_level").as<std::string>());
+    OVMS_LogLevel_enum logLevel;
+    if (cliLogLevel == "TRACE") {
+        logLevel = OVMS_LOG_TRACE;
+    } else if (cliLogLevel == "DEBUG") {
+        logLevel = OVMS_LOG_DEBUG;
+    } else if (cliLogLevel == "INFO") {
+        logLevel = OVMS_LOG_INFO;
+    } else if (cliLogLevel == "WARN") {
+        logLevel = OVMS_LOG_WARNING;
+    } else if (cliLogLevel == "ERROR") {
+        logLevel = OVMS_LOG_ERROR;
+    } else {
+        std::cout << __LINE__ << std::endl;
+        return EX_USAGE;
+    }
+    OVMS_ServerGeneralOptionsSetLogLevel(go, logLevel);
+    OVMS_ServerMultiModelOptionsSetConfigPath(mmo, "/ovms/src/test/c_api/config_standard_dummy.json");
+
+    OVMS_Status* res = OVMS_ServerStartFromConfigurationFile(srv, go, mmo);
+
+    if (res) {
+        // TODO: Better error handling?
+        fprintf(stderr, "Error starting the server\n");
+        OVMS_ServerDelete(srv);
+        OVMS_ServerMultiModelOptionsDelete(mmo);
+        OVMS_ServerGeneralOptionsDelete(go);
+        return 1;
+    }
+
+    std::cout << "Server ready for inference" << std::endl;
+
+    ///////////////////////
+    // model parameters
+    ///////////////////////
+    std::string modelName(cliparser.result->operator[]("model_name").as<std::string>());
+    uint64_t  modelVersion(cliparser.result->operator[]("model_version").as<uint64_t>());
```

Comment:
Can we pass 0 and expect it will be treated as default version?

Reply:
Default is set in cxxopt options (1)